### DISCUSSION TITLE
Alias of Employee

### DISCUSSION THREAD

**Discussion by pponec** [vincentlauvlwj/Ktorm#424] [Q&A] [answered]
Hi, the `Table.aliased()` method called on the `Employees` object returns an result type of `Table<Employee>`, but I would expect a new instance of the `Employee` class. Because of this, the alias does not provide column names. Where is the error?

```kotlin
    fun test() {
        val aliased = Employees.aliased("e") // A result type of Table<Employee>
        var employeeName = Employees.name // OK
        var aliasedName = aliased.name // Compilation error: Unresolved reference !!!
    }
```

The `Employee` class is defined by:

```kotlin
object Employees : Table<Employee>("employee") {
    val id = long("id").primaryKey().bindTo { it.id }
    val name = varchar("name").bindTo { it.name }
}
```

I expected the usage described here in this tutorial:
https://www.ktorm.org/en/joining.html#Self-Joining-amp-Table-Aliases

**Comment by vincentlauvlwj** [comment, marked as answer]
See the doc https://www.ktorm.org/en/joining.html#Self-Joining-amp-Table-Aliases

![image](https://user-images.githubusercontent.com/10897214/185926488-fc9977ce-a832-4fc7-a47d-a1b2636c5479.png)
